### ISSUE TITLE
Arabic data detection and recognition

### ISSUE THREAD

**Issue by haythemBD** [mindee/doctr#490]
Hello, I want to ask if we can train an end-to-end doct for  model on arabic data detection and recgontion??

**Comment by charlesmindee** [comment]
Hi @haythemBD, 

Thanks for your interest in the lib, for the moment we don't have an arabic vocabulary in our [supported vocabs](https://mindee.github.io/doctr/latest/datasets.html#supported-vocabs). If you wish to train a recognition model on arabic data we first need to integrate (and therefore define) an arabic vocabulary, then it will be easy to train a recognition model with this vocabulary if you have arabic-annoted data (at a word level for the recognition task). 
For the detection part, you could also re-train a detection model on arabic data but it should work well with our pretrained model on european data since it detects patterns at a word-level.

If you wish to help us on that :pray:, you are more than welcome to open a PR to propose an arabic vocabulary, then it will be available to define an arabic recognition model and you will just have to train it on your data :smile: ! Please read the [contributing
](https://github.com/mindee/doctr/blob/main/CONTRIBUTING.md) section to be aware of our guidelines! 

Have a nice day!

**Comment by charlesmindee** [comment]
Hi @mzeidhassan, to define an arabic recognition model we only need a vocab like these ones:
```
VOCABS['portuguese'] = VOCABS['english'] + 'áàâãéêëíïóôõúüçÁÀÂÃÉËÍÏÓÔÕÚÜÇ' + '¡¿'
VOCABS['spanish'] = VOCABS['english'] + 'áéíóúüñÁÉÍÓÚÜÑ' + '¡¿'
VOCABS['german'] = VOCABS['english'] + 'äöüßÄÖÜẞ'
```

From there,  if you have an arabic dataset (arabic pictures of words + the corresponding annotations) you can train a recognition model fully operational and even plug it onto a one of our pretrained detection model.

Have a nice day!

**Comment by fg-mindee** [comment]
Small precision @mzeidhassan @haythemBD : if any of you is willing to make a small PR to add the arabic entry in the VOCABS, that would be very helpful :)

**Comment by mzeidhassan** [comment]
Thanks @charlesmindee and @fg-mindee 
I hope this one works.
https://github.com/mindee/doctr/pull/514

**Comment by fg-mindee** [comment]
Closed by #502 & #514

@haythemBD you can find the instructions to train on your arabic dataset here: https://github.com/mindee/doctr/tree/main/references/recognition
And you only have to specify the `--vocab` argument to select the new arabic vocab :)

**Comment by fg-mindee** [comment]
Closed by #502 & #514

@haythemBD you can find the instructions to train on your arabic dataset here: https://github.com/mindee/doctr/tree/main/references/recognition
And you only have to specify the `--vocab` argument to select the new arabic vocab :)
